pkg/data/management/role_data.go

```diff
@@ -637,6 +665,29 @@ func BootstrapAdmin(management *wrangler.Context) (string, error) {
 	return adminName, nil
 }
 
+func addPrincipalLabelToAdmin(admin *v3.User, management *wrangler.Context) (*v3.User, error) {
+	// admin may not have a principal yet. To get around a complex wait loop, gen the principal name as the provider does
+	key, value := getAdminPrincipalLabel(admin)
+	var err error
+	if _, ok := admin.Labels[key]; !ok {
+		newAdmin := admin.DeepCopy()
+		newAdmin.Labels[key] = value
+		newAdmin, err = management.Mgmt.User().Update(newAdmin)
+		return newAdmin, err
+	}
+	return admin, err
+}
+
+func getAdminPrincipalLabel(admin *v3.User) (string, string) {
+	// admin may not have a principal yet. To get around a complex wait loop, gen the principal name as the provider does
```

Comment:
Why wouldn't it have it yet? If there's another component doing that, maybe this logic should be moved there?

Reply:
So in testing, `admin.PrincipalIds` was consistently nil at this point in the boot process. My guess is that the local auth provider is adding the principal when you try to login as the user but I'm not really 100% sure about that. 

That being said, given how we handle other users (adding the labels at create time) I think it makes the most sense to add the labels to this user where it is created rather than trying to stint this in later on. But if you disagree with me let me know and I can look into it further.